js/locale/model-locale-resolver.js

```diff
@@ -27,128 +27,70 @@ export class ModelLocaleResolverPlugin extends RipeCommonsPlugin {
-        color && value.push(`colors.${color}`);
-        value = value.concat(suffixes);
-        return this.localeModel(value, {
+        const ripe = this.ripeProvider.ripe;
+        return ripe.localeColor(color, this.localePlugin, {
+            brand: this.brand,
+            model: this.model,
+            part: part,
+            material: material,
             locale: locale,
-            defaultValue: defaultValue
+            defaultValue: defaultValue,
+            prefixes: prefixes,
+            suffixes: suffixes
         });
     }
 
     localeMaterial(
         material,
         { part = null, locale = null, defaultValue = null, prefixes = [], suffixes = [] } = {}
     ) {
-        let value = [];
-        value = value.concat(prefixes);
-        part && material && value.push(`materials.${part}.${material}`);
-        material && value.push(`materials.${material}`);
-        value = value.concat(suffixes);
-        return this.localeModel(value, {
+        const ripe = this.ripeProvider.ripe;
+        return ripe.localeMaterial(material, this.localePlugin, {
+            brand: this.brand,
+            model: this.model,
+            part: part,
             locale: locale,
-            defaultValue: defaultValue
+            defaultValue: defaultValue,
+            prefixes: prefixes,
+            suffixes: suffixes
         });
     }
 
     localePart(part, { locale = null, defaultValue = null, prefixes = [], suffixes = [] } = {}) {
-        let value = [];
-        value = value.concat(prefixes);
-        part && value.push(`parts.${part}`);
-        value = value.concat(suffixes);
-        return this.localeModel(value, {
+        const ripe = this.ripeProvider.ripe;
+        return ripe.localePart(part, this.localePlugin, {
+            brand: this.brand,
+            model: this.model,
             locale: locale,
-            defaultValue: defaultValue
+            defaultValue: defaultValue,
+            prefixes: prefixes,
+            suffixes: suffixes
         });
     }
 
     localeProperty(
         name,
         { type = null, locale = null, defaultValue = null, prefixes = [], suffixes = [] } = {}
     ) {
-        let value = [];
-        value = value.concat(prefixes);
-        name && value.push(`properties.${name}`);
-        type && name && value.push(`properties.${type}.${name}`);
-        value = value.concat(suffixes);
-        return this.localeModel(value, {
+        const ripe = this.ripeProvider.ripe;
+        return ripe.localeProperty(name, this.localePlugin, {
+            brand: this.brand,
+            model: this.model,
+            type: type,
             locale: locale,
-            defaultValue: defaultValue
+            defaultValue: defaultValue,
+            prefixes: prefixes,
+            suffixes: suffixes
         });
     }
 
-    localeModel(value, { locale = null, defaultValue = null, ...options } = {}) {

```

Comment:
You need to keep this method, it's public API and might be used from other places.